R/test-threads.R

```diff
@@ -330,4 +298,63 @@ delete_models <- function(.mods, .tags = "test threads", .force = FALSE){
 }
 
 
+#' Parse model files and overwrite estimation iterations
+#'
+#' @param .mods a bbi model object or list of model objects. Generally created by `test_threads()`.
```

Comment:
Is this correct to say it can be either a model object or list of model objects? It looks like `.mods` is unconditionally processed as a list. If there is a mismatch, I think the code is fine and the doc description should be updated.

Reply:
Yeah I was thinking about this actually. I think it actually does need to take in a single model as well. I just had the thought, that if users pipe in `.threads = 4` instead of `.treads = c(2,4)`, I think the map call will fail. Will likely add the block I have in some of the other functions:
```r
  if (inherits(.mods, "bbi_model")) {
    .mods <- list(.mods)
  }
```

Reply:
> . I think it actually does need to take in a single model as well. I just had the thought, that if users pipe in .threads = 4 instead of .threads = c(2,4), I think the map call will fail.

I don't see why you think that.  Even for a 1 item vector, map will return a list.

```
> purrr::map(1, ~ .x)
[[1]]
[1] 1
```

I'm still not spotting when this helper function wouldn't receive a list.

Reply:
Oh yeah you're right. I was thinking `length(.mods) = 1`, but that's obviously irrelevant to the class. Ill change the docs instead